packages/dashboard-core-plugins/src/panels/CommandHistoryPanel.tsx

```diff
@@ -1,29 +1,64 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { Component, RefObject } from 'react';
 import { connect } from 'react-redux';
-import { ContextActions } from '@deephaven/components';
-import { CommandHistory, SHORTCUTS } from '@deephaven/console';
-import { GLPropTypes } from '@deephaven/dashboard';
+import { ContextAction, ContextActions } from '@deephaven/components';
+import {
+  CommandHistory,
+  CommandHistoryStorage,
+  SHORTCUTS,
+  CommandHistorySettings,
+  CommandHistoryTable,
+} from '@deephaven/console';
 import Log from '@deephaven/log';
-import { getCommandHistoryStorage } from '@deephaven/redux';
-import { Pending } from '@deephaven/utils';
+import { getCommandHistoryStorage, RootState } from '@deephaven/redux';
+import { assertNotNull, Pending } from '@deephaven/utils';
+import { Container, EventEmitter } from '@deephaven/golden-layout';
+import { IdeSession } from '@deephaven/jsapi-shim';
 import { ConsoleEvent, NotebookEvent } from '../events';
 import './CommandHistoryPanel.scss';
 import Panel from './Panel';
 import { getDashboardSessionWrapper } from '../redux';
 
 const log = Log.module('CommandHistoryPanel');
 
-class CommandHistoryPanel extends Component {
+interface CommandHistoryPanelProps {
+  glContainer: Container;
+  glEventHub: EventEmitter;
+  panelState: Record<string, never>;
+  session: IdeSession;
+  sessionId: string;
+  language: string;
+  commandHistoryStorage: CommandHistoryStorage;
+}
+
+interface CommandHistoryPanelState {
+  panelState: Record<string, never>;
+  session?: IdeSession;
+  sessionId?: string;
+  language?: string;
+  contextActions: ContextAction[];
+  table?: CommandHistoryTable;
+}
+
+class CommandHistoryPanel extends Component<
+  CommandHistoryPanelProps,
+  CommandHistoryPanelState
+> {
+  static defaultProps = {
+    panelState: null,
+    session: null,
+    sessionId: null,
+    language: null,
+  };
```

Comment:
All these props should just stay as `undefined` if not set, meaning you don't need to define any `defaultProps`:
```suggestion
```